PIMS/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,152 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Services.Helper;
+using ServicesContract.Model;
+using ServicesContracts;
+using ServicesContracts.Model;
+using System;
+using System.Threading.Tasks;
+
+namespace Assignment_1.Controllers
+{
+    [Route("/products")]
+    public class ProductsController : Controller
+    {
+        private readonly IProducts _products;
+        private readonly ExportExcel _exportExcel;
+
+        public ProductsController(IProducts products, ExportExcel exportExcel)
+        {
+            _products = products;
+            _exportExcel = exportExcel;
+        }
+
+        [HttpGet("/")]
+        public async Task<string> Home()
+        {
+            return "Hello World";
+        }
+
+
+        /// <summary>
+        /// Endpoint to add a new product.
+        /// </summary>
+        /// <param name="product">The model containing product details.</param>
+        /// <returns>Returns the result of adding the product.</returns>
+       
+        [HttpPost("add")]
+        public async Task<IActionResult> AddProducts([FromBody] ProductsModel product)
+        {
+            try
+            {
+                var result = await _products.AddProducts(product);
+                if (result)
+                {
+                    return Ok("Product added");
+                }
+                return BadRequest("Product not added");
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+
+        /// <summary>
+        /// Endpoint to retrieve all products.
+        /// </summary>
+        /// <returns>Returns the list of all products.</returns>
+       
+        [HttpGet("getall")]
+        public async Task<IActionResult> GetProducts()
+        {
+            try
+            {
+                var result = await _products.GetProducts();
+                return Ok(result);
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+
+        /// <summary>
+        /// Endpoint to delete a product by ID.
+        /// </summary>
+        /// <param name="productId">The ID of the product to delete.</param>
+        /// <returns>Returns the result of deleting the product.</returns>
+        
+        [HttpDelete("delete")]
+        public async Task<IActionResult> DeleteProduct([FromQuery] int productId)
+        {
+            try
+            {
+                var result = await _products.DeleteProduct(productId);
+                if (result)
+                {
+                    return Ok("Product deleted successfully");
+                }
+                return NotFound("Product not found");
```

Comment:
Returned Appropriate HTTP Status Codes, Like NoContent (204) for Delete Operations
Why: Returning appropriate HTTP status codes provides clients with accurate information about the result of their request. For example, a successful delete operation should return 204 No Content instead of 200 OK.

The Ok status code was replaced with NoContent for successful delete operations, as no content is returned in the response.

Reply:
Resolved